### PR TITLE
input protocol & domain forward / cloudfront / whatever magic

### PR DESCRIPTION
**Pull request by julesjanssen**

### CODE REVIEW

File: classes/input.php [hunk 1]
```diff
@@ -192,7 +192,9 @@ public static function real_ip($default = '0.0.0.0')
 	public static function protocol()
 	{
 		if ((static::server('HTTPS') !== null and static::server('HTTPS') != 'off')
-			or (static::server('HTTPS') === null and static::server('SERVER_PORT') == 443))
+			or (static::server('HTTPS') === null and static::server('SERVER_PORT') == 443)
+			or (static::server('HTTP_X_FORWARDED_PROTO') !== null and static::server('HTTP_X_FORWARDED_PROTO') != off)
```
Comment (jschreuder) [inline classes/input.php]: Did you test this with error reporting on?

Comment (julesjanssen) [inline classes/input.php]: Ah, crap, forgot to include that ' commit.
I can't add that commit to this pull request for some reason though.

**Comment by WanWizard** [comment]
i'm not sure this sort of app specific settings should belong in the core. We had the same dicussion recently about reverse proxies, which need similar settings.

I think if you need something like this, deal with it on the app side, for example by setting $_SERVER['HTTPS'] to 'on' based on the HTTP_X_FOWARDED_PROTO value, or whatever setting you need...

**Comment by julesjanssen** [comment]
Agreed. I just wasn't sure how 'app specific' this is and wanted to let you decide. 
Thanks.

**Comment by crynobone** [comment]
Would be helpful if we can setup a knowledge base center for app's specific configuration.

**Pull request closed without merging**